.github/workflows/test.yml

```diff
@@ -18,6 +18,7 @@ jobs:
   run-unit-tests:
     name: Unit Tests
     runs-on: ubuntu-latest
+    timeout-minutes: 30 # TODO: is it low enough?
```

Comment:
did you reached timeout with this workflow?

Reply:
i don't think it should be more than 10 minutes.
if it more then that we need to rethink how we running the tests

Reply:
i don't think it should be more than 10 minutes.
if it more then that we need to rethink how we running the tests

Reply:
I have observed once that the job were taking very long time. I thought that something in unit tests made them hang indefinitely. IMO having safety valve timeout is a good idea. I will change it into 10m